sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java

```diff
@@ -371,29 +358,34 @@ public String getFriendlyName() {
   public List<Run> list(Consumer<String> errorHandler, int offset, int limit, boolean sortDir, String sortCol,
       PaginationFilter... filter)
       throws IOException {
-    List<Run> runs = HibernatePaginatedDataSource.super.list(errorHandler, offset, limit, sortDir, sortCol, filter);
+    List<Run> runs = JpaCriteriaPaginatedDataSource.super.list(errorHandler, offset, limit, sortDir, sortCol, filter);
     if (runs.isEmpty()) {
       return runs;
     }
 
-    @SuppressWarnings("unchecked")
-    List<Object[]> results = currentSession().createCriteria(Run.class)
-        .createAlias("runPositions", "position")
-        .createAlias("position.container", "container")
-        .createAlias("container.partitions", "partition")
-        .createAlias("partition.pool", "pool")
-        .createAlias("pool.poolElements", "element")
-        .createAlias("element.aliquot", "aliquot")
-        .createAlias("aliquot.parentLibrary", "library")
-        .createAlias("library.parentSample", "sample")
-        .createAlias("sample.parentProject", "project")
-        .setProjection(Projections.distinct(
-            Projections.projectionList()
-                .add(Projections.property("id"))
-                .add(Projections.property("project.code"))
-                .add(Projections.property("project.name"))))
-        .add(Restrictions.in("id", runs.stream().map(Run::getId).toArray()))
-        .list();
+    QueryBuilder<Object[], Run> builder = new QueryBuilder<>(currentSession(), Run.class, Object[].class);
+    Join<Run, RunPosition> position = builder.getJoin(builder.getRoot(), Run_.runPositions);
+    Join<RunPosition, SequencerPartitionContainerImpl> container = builder.getJoin(position, RunPosition_.container);
+    Join<SequencerPartitionContainerImpl, PartitionImpl> partition =
+        builder.getJoin(container, SequencerPartitionContainerImpl_.partitions);
+    Join<PartitionImpl, PoolImpl> pool = builder.getJoin(partition, PartitionImpl_.pool);
+    Join<PoolImpl, PoolElement> element = builder.getJoin(pool, PoolImpl_.poolElements);
+    Join<PoolElement, ListLibraryAliquotView> aliquot = builder.getJoin(element, PoolElement_.aliquot);
+    Join<ListLibraryAliquotView, ParentLibrary> library =
+        builder.getJoin(aliquot, ListLibraryAliquotView_.parentLibrary);
+    Join<ParentLibrary, ParentSample> sample = builder.getJoin(library, ParentLibrary_.parentSample);
+    Join<ParentSample, ParentProject> project = builder.getJoin(sample, ParentSample_.parentProject);
+
+    In<Long> inClause = builder.getCriteriaBuilder().in(builder.getRoot().get(Run_.runId));
+    List<Long> ids = runs.stream().map(Run::getId).toList();
+    for (Long id : ids) {
+      inClause.value(id);
+    }
```

Comment:
Can avoid creating the `List<Long>` by looping through the runs and adding value `run.getId()` instead

Reply:
Left `List<Long>` in due to the parameters `addInPredicate` takes in QueryBuilder